### ISSUE TITLE
Make publish-conventions more strongly typed

### ISSUE THREAD

**Issue by breedx-splk** [open-telemetry/opentelemetry-android#273]
In this PR comment:
https://github.com/open-telemetry/opentelemetry-android/pull/272/files#r1521368807

It was noted that the publishing could have been broken due to the project restructuring. We could probably benefit from changing the code in the publish conventions to be more explicit in order to prevent breaks like this in the future.

**Comment by marandaneto** [comment]
I think it's not needed for v1 since it is not a behavior or API surface breakage.

**Comment by LikeTheSalad** [comment]
> I think it's not needed for v1 since it is not a behavior or API surface breakage.

True, making them strongly typed can wait. I think I marked this as required because I was also thinking about defining what the instrumentation URIs format should look like, like `io.opentelemetry.android.instrumentation:[project-name]` or `io.opentelemetry.android:instrumentation-[project-name]`. It's probably better to create a new issue that's focused more on that part instead.

**Comment by LikeTheSalad** [comment]
I've created this issue to discuss it: https://github.com/open-telemetry/opentelemetry-android/issues/533

**Comment by fractalwrench** [comment]
@breedx-splk I _think_ this is addressed by #533 but will leave it up to you to confirm + close this issue.

**Comment by breedx-splk** [comment]
I _think_ so? Let's close it and reopen if it becomes an issue again. Thanks for the nudge.
